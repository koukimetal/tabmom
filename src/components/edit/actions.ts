import { CronRule } from './../table/actions';
import { combineReducers } from 'redux';
const UPDATE_URL = '@edit/UPDATE_URL';
const UPDATE_NAME = '@edit/UPDATE_NAME';
const UPDATE_PERIOD = '@edit/UPDATE_PERIOD';
const UPDATE_ACTIVE = '@edit/UPDATE_ACTIVE';

const OPEN_MODAL_CREATE = '@edit/OPEN_MODAL_CREATE';
const OPEN_MODAL_EDIT = '@edit/OPEN_MODAL_EDIT';
const CLOSE_MODAL = '@edit/CLOSE_MODAL';

const UPDATE_DELETE_FLAG = '@edit/UPDATE_DELETE_FLAG';

interface NameAction {
    type: typeof UPDATE_NAME;
    name: string;
}

interface UrlAction {
    type: typeof UPDATE_URL;
    url: string;
}

interface PeriodAction {
    type: typeof UPDATE_PERIOD;
    period: string;
}

interface ActiveAction {
    type: typeof UPDATE_ACTIVE;
    active: boolean;
}

interface OpenCreateAction {
    type: typeof OPEN_MODAL_CREATE;
}

interface OpenEditAction {
    type: typeof OPEN_MODAL_EDIT;
    rule: CronRule;
    index: number;
}

interface DeleteFlagAction {
    type: typeof UPDATE_DELETE_FLAG;
    flag: boolean;
}

interface CloseAction {
    type: typeof CLOSE_MODAL;
}

export enum ModalMode {
    CREATE,
    EDIT,
    CLOSED,
}

type EditModalAction =
    | NameAction
    | UrlAction
    | PeriodAction
    | OpenCreateAction
    | CloseAction
    | ActiveAction
    | DeleteFlagAction
    | OpenEditAction;

export interface EditModalState {
    period: string;
    name: string;
    url: string;
    active: boolean;
    mode: ModalMode;
    editIndex: number;
    deleteFlag: boolean;
}

const modalModeReducer = (state = ModalMode.CLOSED, action: EditModalAction) => {
    switch (action.type) {
        case OPEN_MODAL_CREATE:
            return ModalMode.CREATE;
        case OPEN_MODAL_EDIT:
            return ModalMode.EDIT;
        case CLOSE_MODAL:
            return ModalMode.CLOSED;
        default:
            return state;
    }
};

const editIndexReducer = (state = -1, action: EditModalAction) => {
    switch (action.type) {
        case OPEN_MODAL_CREATE:
            return -1;
        case OPEN_MODAL_EDIT:
            return action.index;
        case CLOSE_MODAL:
            return -1;
        default:
            return state;
    }
};

const defaultReducer = <T>(state: T, initialValue: T, action: EditModalAction) => {
    switch (action.type) {
        case CLOSE_MODAL:
            return initialValue;
        default:
            return state;
    }
};

const deleteFlagReducer = (state = false, action: EditModalAction) => {
    switch (action.type) {
        case UPDATE_DELETE_FLAG:
            return action.flag;
        default:
            return defaultReducer(state, false, action);
    }
};

const nameReducer = (state = '', action: EditModalAction) => {
    switch (action.type) {
        case UPDATE_NAME:
            return action.name;
        case OPEN_MODAL_EDIT:
            return action.rule.name;
        default:
            return defaultReducer(state, '', action);
    }
};

const urlReducer = (state = '', action: EditModalAction) => {
    switch (action.type) {
        case UPDATE_URL:
            return action.url;
        case OPEN_MODAL_EDIT:
            return action.rule.url;
        default:
            return defaultReducer(state, '', action);
    }
};

const periodReducer = (state = '', action: EditModalAction) => {
    switch (action.type) {
        case UPDATE_PERIOD:
            return action.period;
        case OPEN_MODAL_EDIT:
            return action.rule.period.toString();
        default:
            return defaultReducer(state, '', action);
    }
};

const activeReducer = (state = true, action: EditModalAction) => {
    switch (action.type) {
        case UPDATE_ACTIVE:
            return action.active;
        case OPEN_MODAL_EDIT:
            return action.rule.active;
        default:
            return defaultReducer(state, true, action);
    }
};

export const openModal = (): OpenCreateAction => {
    return {
        type: OPEN_MODAL_CREATE,
    };
};

export const editModal = (rule: CronRule, index: number): OpenEditAction => {
    return {
        type: OPEN_MODAL_EDIT,
        rule,
        index,
    };
};

export const closeModal = (): CloseAction => {
    return {
        type: CLOSE_MODAL,
    };
};

export const updateName = (name: string): NameAction => {
    return {
        type: UPDATE_NAME,
        name,
    };
};
export const updateUrl = (url: string): UrlAction => {
    return {
        type: UPDATE_URL,
        url,
    };
};
export const updatePeriod = (period: string): PeriodAction => {
    return {
        type: UPDATE_PERIOD,
        period,
    };
};

export const updateActive = (active: boolean): ActiveAction => {
    return {
        type: UPDATE_ACTIVE,
        active,
    };
};

export const updateDeleteFlag = (flag: boolean): DeleteFlagAction => {
    return {
        type: UPDATE_DELETE_FLAG,
        flag,
    };
};

export const editModalReducer = combineReducers<EditModalState, EditModalAction>({
    name: nameReducer,
    url: urlReducer,
    period: periodReducer,
    mode: modalModeReducer,
    active: activeReducer,
    editIndex: editIndexReducer,
    deleteFlag: deleteFlagReducer,
});
