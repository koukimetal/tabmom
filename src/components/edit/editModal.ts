import { getSimpleReducerAndActionCreator, SimpleAction } from './../../utilities/simpleReducer';
import { combineReducers } from 'redux';
const UPDATE_URL = '@edit/UPDATE_URL'
const UPDATE_NAME = '@edit/UPDATE_NAME'
const UPDATE_PERIOD = '@edit/UPDATE_PERIOD'

export type EditModalTypes = SimpleAction<string>;

export interface EditModalState {
    period: string
    name: string
    url: string
}

const nameRAC = getSimpleReducerAndActionCreator('', UPDATE_NAME);
const urlRAC  = getSimpleReducerAndActionCreator('', UPDATE_URL);
const periodRAC  = getSimpleReducerAndActionCreator('', UPDATE_PERIOD);

export const updateName = nameRAC.actionCreator;
export const updateURL = urlRAC.actionCreator;
export const updatePeriod = periodRAC.actionCreator;

const editModalReducer = combineReducers<EditModalState, EditModalTypes>({
    name: nameRAC.reducer,
    url: urlRAC.reducer,
    period: periodRAC.reducer
});
