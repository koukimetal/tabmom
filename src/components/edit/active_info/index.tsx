import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../store';
import { updateActive, updateOneTime, updateIsWeekSettingActive } from '../actions';
import { createStyles, WithStyles, FormGroup, FormControlLabel, Checkbox } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
import { WeekSetting } from './week_setting';

const styles = () => createStyles({});

interface DispatchProps {
    updateActive: typeof updateActive;
    updateOneTime: typeof updateOneTime;
    updateIsWeekSettingActive: typeof updateIsWeekSettingActive;
}

interface StateProps {
    active: boolean;
    oneTime: boolean;
    isWeekSettingActive: boolean;
}

interface Props extends DispatchProps, StateProps, WithStyles<typeof styles> {}

class ActiveInfoInner extends React.Component<Props> {
    private toggleActive = () => {
        this.props.updateActive(!this.props.active);
    };
    private toggleOneTime = () => {
        this.props.updateOneTime(!this.props.oneTime);
    };

    private toggleWeekSettingActive = () => {
        this.props.updateIsWeekSettingActive(!this.props.isWeekSettingActive);
    };

    public render() {
        const { active, isWeekSettingActive, oneTime } = this.props;
        return (
            <>
                <div>
                    <FormGroup row>
                        <FormControlLabel
                            control={<Checkbox checked={active} onChange={this.toggleActive} value={active} />}
                            label="Active"
                        />
                        <FormControlLabel
                            control={<Checkbox checked={oneTime} onChange={this.toggleOneTime} value={oneTime} />}
                            label="OneTime"
                        />
                        <FormControlLabel
                            control={
                                <Checkbox
                                    checked={isWeekSettingActive}
                                    onChange={this.toggleWeekSettingActive}
                                    value={isWeekSettingActive}
                                />
                            }
                            label="UseWeekSetting"
                        />
                    </FormGroup>
                </div>
                {isWeekSettingActive && <WeekSetting />}
            </>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    active: state.edit.active,
    oneTime: state.edit.oneTime,
    isWeekSettingActive: state.edit.isWeekSettingActive,
});

export const ActiveInfo = connect<StateProps, DispatchProps>(
    mapStateToProps,
    {
        updateActive,
        updateOneTime,
        updateIsWeekSettingActive,
    },
)(withStyles(styles)(ActiveInfoInner));
