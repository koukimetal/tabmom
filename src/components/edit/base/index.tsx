import * as React from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/styles';
import { closeModal } from '../actions';
import { createStyles, WithStyles } from '@material-ui/core';
import { UrlForm } from './url';
import { NameForm } from './name';
import { Close as CloseIcon } from '@material-ui/icons';

const styles = () =>
    createStyles({
        top: {
            display: 'flex',
        },
        close: {
            marginLeft: 'auto',
        },
    });

interface DispatchProps {
    closeModal: typeof closeModal;
}

interface Props extends DispatchProps, WithStyles<typeof styles> {}

class UrlFormInner extends React.Component<Props> {
    private close = () => {
        this.props.closeModal();
    };
    public render() {
        const { classes } = this.props;
        return (
            <>
                <div className={classes.top}>
                    <NameForm />
                    <div className={classes.close}>
                        <CloseIcon onClick={this.close} />
                    </div>
                </div>
                <UrlForm />
            </>
        );
    }
}

export const BaseForm = connect<null, DispatchProps>(
    null,
    {
        closeModal,
    },
)(withStyles(styles)(UrlFormInner));
