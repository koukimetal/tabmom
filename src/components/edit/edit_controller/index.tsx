import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../store';
import { ModalMode } from '../actions';
import { createStyles, WithStyles } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
import { EditDelete } from './delete';
import { SaveButton, CopyButton } from './save';

const styles = () =>
    createStyles({
        root: {
            display: 'flex',
        },
    });

interface StateProps {
    mode: ModalMode;
}

interface Props extends StateProps, WithStyles<typeof styles> {}

class EditControllerInner extends React.Component<Props> {
    public render() {
        const { mode, classes } = this.props;
        return (
            <div className={classes.root}>
                <SaveButton />
                {mode === ModalMode.EDIT && (
                    <>
                        <EditDelete />
                        <CopyButton />
                    </>
                )}
            </div>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    mode: state.edit.mode,
});

export const EditController = connect<StateProps, {}>(
    mapStateToProps,
    {},
)(withStyles(styles)(EditControllerInner));
