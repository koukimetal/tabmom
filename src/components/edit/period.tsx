import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../store';
import {
    EditModalState,
    updatePeriod,
    ModalMode,
    updateStartTime,
    updateEndTime,
    updateCurrent as editUpdateCurrent,
    updateTimeRangeType,
} from './actions';
import { TextField, Theme, createStyles, WithStyles, Select } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
import { TimeRangeType } from '../system/actions';

const styles = (theme: Theme) =>
    createStyles({
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200,
        },
        typeOptions: {
            marginLeft: 5,
        },
    });

interface DispatchProps {
    updatePeriod: typeof updatePeriod;
    editUpdateCurrent: typeof editUpdateCurrent;
    updateStartTime: typeof updateStartTime;
    updateEndTime: typeof updateEndTime;
    updateTimeRangeType: typeof updateTimeRangeType;
}

interface StateProps {
    edit: EditModalState;
}

interface Props extends DispatchProps, StateProps, WithStyles<typeof styles> {}

class EditModalPeriodInner extends React.Component<Props> {
    private changePeriod = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.updatePeriod(event.currentTarget.value);
    };
    private changeCurrent = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.editUpdateCurrent(event.currentTarget.value);
    };

    private changeStartTime = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.updateStartTime(event.currentTarget.value);
    };
    private changeEndTime = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.props.updateEndTime(event.currentTarget.value);
    };
    private changeClockConfigType = (event: React.ChangeEvent<HTMLInputElement>) => {
        const key = event.currentTarget.value as keyof typeof TimeRangeType;
        this.props.updateTimeRangeType(TimeRangeType[key]);
    };

    private renderForOnceADay = () => {
        const { edit, classes } = this.props;

        return (
            <div>
                <TextField
                    label="TimeToTrigger"
                    className={classes.textField}
                    value={edit.clockConfig.startTime}
                    onChange={this.changeStartTime}
                    type="time"
                    margin="normal"
                />
            </div>
        );
    };

    private renderForALL = () => {
        const { edit, classes } = this.props;
        return (
            <div>
                {edit.mode === ModalMode.EDIT && (
                    <TextField
                        label="Current"
                        className={classes.textField}
                        value={edit.current}
                        onChange={this.changeCurrent}
                        margin="normal"
                    />
                )}
                <TextField
                    label="Period"
                    className={classes.textField}
                    value={edit.clockConfig.period}
                    onChange={this.changePeriod}
                    margin="normal"
                />
            </div>
        );
    };

    private renderForMany = () => {
        const { edit, classes } = this.props;
        return (
            <>
                {this.renderForALL()}
                <div>
                    <TextField
                        label="StartTime"
                        className={classes.textField}
                        value={edit.clockConfig.startTime}
                        onChange={this.changeStartTime}
                        type="time"
                        margin="normal"
                    />
                    <TextField
                        label="EndTime"
                        className={classes.textField}
                        value={edit.clockConfig.endTime}
                        type="time"
                        onChange={this.changeEndTime}
                        margin="normal"
                    />
                </div>
            </>
        );
    };

    public render() {
        const { edit, classes } = this.props;
        return (
            <>
                <div className={classes.typeOptions}>
                    <Select
                        native
                        value={edit.clockConfig.type}
                        onChange={this.changeClockConfigType}
                        inputProps={{
                            name: 'RangeType',
                        }}
                    >
                        <option value={TimeRangeType.ALL.toString()}>All</option>
                        <option value={TimeRangeType.MANY.toString()}>Many</option>
                        <option value={TimeRangeType.ONCE.toString()}>Once</option>
                    </Select>
                </div>
                {edit.clockConfig.type === TimeRangeType.ONCE && this.renderForOnceADay()}
                {edit.clockConfig.type === TimeRangeType.ALL && this.renderForALL()}
                {edit.clockConfig.type === TimeRangeType.MANY && this.renderForMany()}
            </>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    edit: state.edit,
});

export const EditModalPeriod = connect<StateProps, DispatchProps>(
    mapStateToProps,
    {
        updatePeriod,
        editUpdateCurrent,
        updateStartTime,
        updateEndTime,
        updateTimeRangeType,
    },
)(withStyles(styles)(EditModalPeriodInner));
