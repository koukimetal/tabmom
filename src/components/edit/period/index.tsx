import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../store';
import { updateTimeRangeType } from '../actions';
import { createStyles, WithStyles, Select, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
import { TimeRangeType } from '../../system/actions';
import { OnceADay } from './once';
import { ManyPeriod } from './many';
import { AllPeriod } from './all';

const styles = () =>
    createStyles({
        typeOptions: {
            marginLeft: 5,
        },
        modeText: {
            marginRight: 5,
        },
    });

interface DispatchProps {
    updateTimeRangeType: typeof updateTimeRangeType;
}

interface StateProps {
    type: TimeRangeType;
}

interface Props extends DispatchProps, StateProps, WithStyles<typeof styles> {}

class EditModalPeriodInner extends React.Component<Props> {
    private changeClockConfigType = (event: React.ChangeEvent<HTMLInputElement>) => {
        const key = event.currentTarget.value as keyof typeof TimeRangeType;
        this.props.updateTimeRangeType(TimeRangeType[key]);
    };

    public render() {
        const { type, classes } = this.props;
        return (
            <>
                <div className={classes.typeOptions}>
                    <Typography variant="subtitle1" component="span" className={classes.modeText}>
                        Mode for a day:
                    </Typography>
                    <Select
                        native
                        value={type}
                        onChange={this.changeClockConfigType}
                        inputProps={{
                            name: 'RangeType',
                        }}
                    >
                        <option value={TimeRangeType.ALL.toString()}>All</option>
                        <option value={TimeRangeType.MANY.toString()}>Many</option>
                        <option value={TimeRangeType.ONCE.toString()}>Once</option>
                    </Select>
                </div>
                {type === TimeRangeType.ONCE && <OnceADay />}
                {type === TimeRangeType.ALL && <AllPeriod />}
                {type === TimeRangeType.MANY && <ManyPeriod />}
            </>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    type: state.edit.clockConfig.type,
});

export const EditModalPeriod = connect<StateProps, DispatchProps>(
    mapStateToProps,
    {
        updateTimeRangeType,
    },
)(withStyles(styles)(EditModalPeriodInner));
