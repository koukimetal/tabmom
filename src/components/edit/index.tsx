import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../store';
import { closeModal, ModalMode } from './actions';
import Modal from '@material-ui/core/Modal';
import { Theme, createStyles, WithStyles, Paper } from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
import { EditSkipInfo } from './skip_info';
import { EditModalPeriod } from './period';
import { BaseForm } from './base';
import { ActiveInfo } from './active_info';
import { EditController } from './edit_controller';

const styles = (theme: Theme) =>
    createStyles({
        root: {
            padding: theme.spacing(1),
        },
    });

interface DispatchProps {
    closeModal: typeof closeModal;
}

interface StateProps {
    mode: ModalMode;
}

interface Props extends DispatchProps, StateProps, WithStyles<typeof styles> {}

class EditModalInner extends React.Component<Props> {
    private close = () => {
        this.props.closeModal();
    };

    public render() {
        const { mode, classes } = this.props;
        return (
            <Modal open={mode !== ModalMode.CLOSED} onClose={this.close}>
                <>
                    <Paper className={classes.root}>
                        <BaseForm />
                        <EditModalPeriod />
                        <ActiveInfo />
                        <EditSkipInfo />
                        <EditController />
                    </Paper>
                </>
            </Modal>
        );
    }
}

const mapStateToProps = (state: AppState) => ({
    mode: state.edit.mode,
});

export const EditModal = connect<StateProps, DispatchProps>(
    mapStateToProps,
    {
        closeModal,
    },
)(withStyles(styles)(EditModalInner));
