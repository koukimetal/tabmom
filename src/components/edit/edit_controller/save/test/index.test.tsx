import * as React from 'react';
import { EditModalState } from '../../../../edit/actions';
import { TimeRangeType } from '../../../../system/actions';
import { shallow } from 'enzyme';
import { SaveButtonInner, SaveCopyBaseProps, CopyButtonInner } from '..';

/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-object-literal-type-assertion */

describe('Test save copy', () => {
    let defaultValue: SaveCopyBaseProps = null;
    const closeModal = jest.fn();
    const addRule = jest.fn();
    const updateRule = jest.fn();
    const deleteCurrent = jest.fn();
    const systemUpdateCurrent = jest.fn();

    beforeEach(() => {
        defaultValue = {
            closeModal,
            addRule,
            updateRule,
            deleteCurrent,
            systemUpdateCurrent,
            edit: null as EditModalState,
            validInput: true,
            classes: {} as any,
        };
        (Object.keys(defaultValue) as (keyof typeof defaultValue)[]).forEach(key => {
            if (typeof defaultValue[key] === 'function') {
                (defaultValue[key] as jest.Mock).mockClear();
            }
        });
    });

    test('test convert time', () => {
        const wd = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        expect(wd.instance().convertTimeToNumber('23:59')).toEqual(60 * 24 - 1);
    });

    test('test getClockConfig', () => {
        const wd = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        expect(
            wd.instance().getClockConfig({
                type: TimeRangeType.ALL,
                period: '1',
            }),
        ).toMatchSnapshot();
        expect(
            wd.instance().getClockConfig({
                type: TimeRangeType.ONCE,
                startTime: '00:00',
            }),
        ).toMatchSnapshot();
        expect(
            wd.instance().getClockConfig({
                type: TimeRangeType.MANY,
                period: '3',
                startTime: '00:00',
                endTime: '01:23',
            }),
        ).toMatchSnapshot();
    });

    test('test getRule', () => {
        defaultValue.edit = {
            clockConfig: {
                type: TimeRangeType.MANY,
                period: '3',
                startTime: '00:00',
                endTime: '01:23',
            },
            isWeekSettingActive: true,
            isSkipInfoActive: true,
            skipInfo: {
                ignorePinned: true,
                match: 'aaa',
            },
            weekSetting: [true, true, true, true, true, true, true],
        } as EditModalState;
        const wd1 = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        expect(wd1.instance().getRule('id')).toMatchSnapshot();
        defaultValue.edit.isSkipInfoActive = false;
        defaultValue.edit.isWeekSettingActive = false;
        const wd2 = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        expect(wd2.instance().getRule('id')).toMatchSnapshot();
    });

    test('test update', () => {
        defaultValue.edit = {
            clockConfig: {
                type: TimeRangeType.MANY,
                period: '3',
                startTime: '00:00',
                endTime: '01:23',
            },
            current: '2',
        } as EditModalState;
        const wd1 = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);

        // need period with update
        wd1.instance().update('id1');
        expect(systemUpdateCurrent.call.length).toEqual(1);
        expect(updateRule.call.length).toEqual(1);

        systemUpdateCurrent.mockClear();

        // need period with create
        wd1.instance().update(null);
        expect(systemUpdateCurrent.call.length).toEqual(1);
        expect(addRule.call.length).toEqual(1);

        // no need period
        defaultValue.edit.clockConfig.type = TimeRangeType.ONCE;
        const wd2 = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        wd2.instance().update(null);
        expect(deleteCurrent.call.length).toEqual(1);
    });

    test('test save', () => {
        defaultValue.edit = { targetId: 'id' } as EditModalState;
        const wd = shallow<SaveButtonInner>(<SaveButtonInner {...defaultValue} />);
        wd.instance().update = jest.fn();
        wd.instance().save({ preventDefault: jest.fn() } as any);
        expect(closeModal.call.length).toEqual(1);
    });

    test('test copy', () => {
        const wd = shallow<CopyButtonInner>(<CopyButtonInner {...defaultValue} />);
        wd.instance().update = jest.fn();
        wd.instance().copy();
        expect(closeModal.call.length).toEqual(1);
    });
});
