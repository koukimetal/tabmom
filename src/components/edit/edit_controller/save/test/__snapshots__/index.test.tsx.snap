// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test save copy test getClockConfig 1`] = `
Object {
  "period": 1,
  "type": "ALL",
}
`;

exports[`Test save copy test getClockConfig 2`] = `
Object {
  "startTime": 0,
  "type": "ONCE",
}
`;

exports[`Test save copy test getClockConfig 3`] = `
Object {
  "endTime": 83,
  "period": 3,
  "startTime": 0,
  "type": "MANY",
}
`;

exports[`Test save copy test getRule 1`] = `
Object {
  "active": undefined,
  "clockConfig": Object {
    "endTime": 83,
    "period": 3,
    "startTime": 0,
    "type": "MANY",
  },
  "id": "id",
  "name": undefined,
  "oneTime": undefined,
  "skipInfo": Object {
    "ignorePinned": true,
    "match": "aaa",
  },
  "url": undefined,
  "weekSetting": Array [
    true,
    true,
    true,
    true,
    true,
    true,
    true,
  ],
}
`;

exports[`Test save copy test getRule 2`] = `
Object {
  "active": undefined,
  "clockConfig": Object {
    "endTime": 83,
    "period": 3,
    "startTime": 0,
    "type": "MANY",
  },
  "id": "id",
  "name": undefined,
  "oneTime": undefined,
  "url": undefined,
}
`;
